analytical_engine/core/fragment/de_mutable_csr.h

```diff
@@ -0,0 +1,780 @@
+/** Copyright 2020 Alibaba Group Holding Limited.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#ifndef ANALYTICAL_ENGINE_CORE_FRAGMENT_DE_MUTABLE_CSR_H_
+#define ANALYTICAL_ENGINE_CORE_FRAGMENT_DE_MUTABLE_CSR_H_
+
+#include <limits>
+#include <map>
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "grape/graph/de_mutable_csr.h"
+
+#include "core/object/dynamic.h"
+
+namespace grape {
+
+template <>
+class DeMutableCSR<
```

Comment:
Could you add some comments?
- for the class 
- for the definition of the dense_threshold